neon: Change the neon configure to use pkg-config instead of xml2-config

xml2-config is broken for neon
if packageconfig libxml2, webdav, zlib is enabled for neon
we get the following configure error in the yocto build

| configure: WebDAV support is enabled
| checking for xml2-config... xml2-config
| ERROR: /usr/bin/xml2-config should not be used, use an alternative such as pkg-config
| ERROR: /usr/bin/xml2-config should not be used, use an alternative such as pkg-config
| ERROR: /usr/bin/xml2-config should not be used, use an alternative such as pkg-config
| checking libxml/xmlversion.h usability... no
| checking libxml/xmlversion.h presence... no
| checking for libxml/xmlversion.h... no
| configure: error: could not find parser.h, libxml installation problem?
| WARNING: exit code 1 from a shell command.

The patch lets configure use pkg-config

Upstream-Status: inappropriate
(Upstream suggests to use latest 0.31 as per the discussion
https://github.com/notroj/neon/discussions/47)

Signed-off-by: Nisha Parrakat <Nisha.Parrakat@kpit.com>
--- a/macros/neon-xml-parser.m4        2008-07-19 23:52:35.000000000 +0200
+++ b/macros/neon-xml-parser.m4        2021-02-15 23:56:59.202751257 +0100
@@ -44,17 +44,17 @@

 dnl Find libxml2: run $1 if found, else $2
 AC_DEFUN([NE_XML_LIBXML2], [
-AC_CHECK_PROG(XML2_CONFIG, xml2-config, xml2-config)
+AC_CHECK_PROG(XML2_CONFIG, pkg-config, pkg-config)
 if test -n "$XML2_CONFIG"; then
-    neon_xml_parser_message="libxml `$XML2_CONFIG --version`"
     AC_DEFINE(HAVE_LIBXML, 1, [Define if you have libxml])
-    # xml2-config in some versions erroneously includes -I/include
-    # in the --cflags output.
-    CPPFLAGS="$CPPFLAGS `$XML2_CONFIG --cflags | sed 's| -I/include||g'`"
-    NEON_LIBS="$NEON_LIBS `$XML2_CONFIG --libs | sed 's|-L/usr/lib ||g'`"
+    PKG_CHECK_MODULES(XML, libxml-2.0 >= 2.4)
+    AC_MSG_NOTICE([libxmlfound CFlags : , ${XML_CFLAGS}])
+    CPPFLAGS="$CPPFLAGS ${XML_CFLAGS}"
+    NEON_LIBS="$NEON_LIBS ${XML_LIBS}"
     AC_CHECK_HEADERS(libxml/xmlversion.h libxml/parser.h,,[
       AC_MSG_ERROR([could not find parser.h, libxml installation problem?])])
     neon_xml_parser=libxml2
+    neon_xml_parser_message="libxml2"
 else
     $1
 fi
